core/src/main/java/org/apache/brooklyn/core/config/BasicConfigInheritance.java

```diff
@@ -170,12 +324,103 @@ public String getConflictResolutionStrategy() {
         return conflictResolutionStrategy;
     }
     
+    @Deprecated /** @deprecated since 0.10.0 when it was introduced, prefer {@link #getLocalDefaultResolvesWithAncestorValue()} */
     public boolean getUseLocalDefaultValue() {
-        return useLocalDefaultValue;
+        return getLocalDefaultResolvesWithAncestorValue();
+    }
+
+    /** see {@link #localDefaultResolvesWithAncestorValue} */
+    public boolean getLocalDefaultResolvesWithAncestorValue() {
+        if (localDefaultResolvesWithAncestorValue==null) {
+            // in case some legacy path is using an improperly deserialized object
+            log.warn("Encountered legacy "+this+" with null localDefaultResolvesWithAncestorValue; transforming", new Throwable("stack trace for legacy "+this));
+            readResolve();
```

Comment:
Are you suspecting the object could be coming from a non-xstream based deserialization? If not, then this is too much on the defensive side.